examples/required-options/index.d.ts

```diff
@@ -0,0 +1,12 @@
+import { Base } from "../../index.js";
+export { Base } from "../../index.js";
+
+declare module "../.." {
+  namespace Base {
+    interface Options {
+      myRequiredUserOption: string;
+    }
+  }
+}
+
+export class MyBase extends Base {}
```

Comment:
Is this the way I'd export a custom Class type if I don't add any plugins?

Reply:
This would work. This would probably also work too:

```ts
export { Base as MyBase } from "../../index.js";
```

...if you don't mind re-using the existing class and not having an explicit new one.

Reply:
This looks right as the `.d.ts` file for the `required-options/index.js` file, yeah. Since `MyClass `extends Base` in the implementation and doesn't add any new files, I don't see anything else you'd need to do.